application-admintools-default/src/main/java/com/xwiki/admintools/internal/wikiSize/InstanceUsage.java

```diff
@@ -0,0 +1,118 @@
+import org.xwiki.wiki.descriptor.WikiDescriptor;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xwiki.admintools.WikiSizeResult;
+import com.xwiki.admintools.internal.PingProvider;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+
+/**
+ * Access info about the size of the existing Wikis.
+ *
+ * @version $Id$
+ */
+@Component(roles = InstanceUsage.class)
+@Singleton
+public class InstanceUsage
+{
+    private static final String TEMPLATE_NAME = "wikiSizeTemplate.vm";
+
+    @Inject
+    private PingProvider pingProvider;
+
+    @Inject
+    private WikiDescriptorManager wikiDescriptorManager;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    @Inject
+    private WikiSizeProvider wikiSizeProvider;
+
+    /**
+     * Get the data in a format given by the associated template.
+     *
+     * @return the rendered template as a {@link String}.
+     */
+    public String renderTemplate()
+    {
+        try {
+            ScriptContext scriptContext = this.scriptContextManager.getScriptContext();
+            boolean found = currentServer.getCurrentServer() != null;
+            scriptContext.setAttribute("found", found, ScriptContext.ENGINE_SCOPE);
+            if (!found) {
+                this.logger.error("Used server not found!");
+                return this.templateManager.render(TEMPLATE_NAME);
+            }
+            List<WikiSizeResult> wikisInfo = getWikisSizeInfo();
+            scriptContext.setAttribute("wikisInfo", wikisInfo, ScriptContext.ENGINE_SCOPE);
+            int numberOfExtensions = pingProvider.getExtensionPing().size();
+            scriptContext.setAttribute("numberOfExtensions", numberOfExtensions, ScriptContext.ENGINE_SCOPE);
+            long totalNumberOfUsers = pingProvider.getUsersPing().getTotal();
+            scriptContext.setAttribute("totalNumberOfUsers", totalNumberOfUsers, ScriptContext.ENGINE_SCOPE);
+            return this.templateManager.render(TEMPLATE_NAME);
+        } catch (Exception e) {
+            this.logger.warn("Failed to render [{}] template. Root cause is: [{}]", TEMPLATE_NAME,
+                ExceptionUtils.getRootCauseMessage(e));
+            return null;
+        }
+    }
+
+    private List<WikiSizeResult> getWikisSizeInfo()
+    {
+        List<WikiSizeResult> result = new ArrayList<>();
+        try {
+            Collection<WikiDescriptor> wikisDescriptors = this.wikiDescriptorManager.getAll();
+            for (WikiDescriptor wikiDescriptor : wikisDescriptors) {
+                result.add(wikiSizeProvider.getWikiSizeInfo(wikiDescriptor));
+            }
+            return result;
+        } catch (Exception e) {
+            logger.warn("There have been issues while gathering info about the size of the Wikis. Root cause is: [{}]",
```

Comment:
`There have been issues while gathering instance usage data. Root cause is: [{}]`